src/injected/web/gm-api.js

```diff
@@ -201,12 +201,24 @@ export function createGmApiProps() {
       });
     },
   };
+  // content scripts can't have sourceURL comment so we need to show the script name explicitly
+  if (bridge.mode === INJECT_CONTENT) {
+    assign(boundProps, {
+      GM_log(...args) {
+        log('log', [this.script.meta.name || 'No name'], ...args);
+      },
+    });
+  }
   // convert to object property descriptors
   [props, boundProps].forEach(target => {
     objectKeys(target).forEach(k => {
       target[k] = propertyFromValue(target[k]);
     });
   });
+  // defining manually to avoid setting toString on the real console.log
+  if (bridge.mode !== INJECT_CONTENT) {
+    props.GM_log = { value: logging.log };
```

Comment:
We need to wrap `console.log` into a new function to avoid `console.log` being modified. (#688)

Reply:
Also I prefer consistent behaviors of `GM_log` no matter it's in page context or content context. Otherwise users have to figure out the difference and learn about `@inject-into`.

Reply:
1. No, we don't need to wrap it since we don't change its toString, I thought the comment I've added on line 218 is obvious but I guess I should improve it?
2. The old internal behavior of GM_log with the "[tags]" is a hack which was used only because the script name wasn't displayed. Since the `content` mode is also a hack it's only natural that two hacks are used together, but normal scripts in page mode don't suffer from this deficiency and use normal logging with a proper script name displayed as the source.

Reply:
On the other hand, GM_log itself is a rarely used thing which I never needed so I don't have a strong opinion. Let's go with your preference.

Reply:
I think I see why you insist to provide the original `console.log` now, it seems that the line info of where `console.log` is exactly called will be displayed on the right. So if we wrap it into a new function, the line info will always be that of `GM_log` definition.

Actually the source URL works also for `content` mode, I'll make a PR for that before merging this one.